### PR TITLE
[Llama] Support hybrid strategy and recompute in llama auto version

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Bug fixes
### PR changes

Others
### Description

[Llama] Support hybrid strategy and recompute in llama auto version

### CODE REVIEW

File: paddlenlp/trainer/training_args.py [hunk 1]
```diff
@@ -1171,15 +1171,19 @@ def is_segment_parallel_supported():
...
                 amp.dtype = "bfloat16" if self.bf16 else "float16"
-                amp.level = self.fp16_opt_level
+                amp.level = self.fp16_opt_level.lower()
                 amp.init_loss_scaling = self.scale_loss
-                amp.custom_black_list = self.amp_custom_black_list
-                amp.custom_white_list = self.amp_custom_white_list
+                amp.custom_black_list = self.amp_custom_black_list if self.amp_custom_black_list is not None else []
+                amp.custom_white_list = self.amp_custom_white_list if self.amp_custom_white_list is not None else []
+                amp.use_fp16_guard = False
+                amp.use_bf16_guard = False
```
Comment (zhiqiu) [inline paddlenlp/trainer/training_args.py]: duplicate line

File: llm/llama/run_auto.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+# just for debug
+
+set -x
+unset CUDA_VISIBLE_DEVICES
+
+# export GLOG_v=4
+export FLAGS_call_stack_level=2
+
+task_name="llama_auto_dp2"
```
Comment (zhiqiu) [inline llm/llama/run_auto.sh]: dp2pp2mp2

File: llm/llama/run_pretrain_auto.py [hunk 3]
```diff
@@ -457,6 +457,9 @@ def main():
         if training_args.bf16:
             dtype = "bfloat16"
 
+    # random.seed(training_args.seed)
```
Comment (zhiqiu) [inline llm/llama/run_pretrain_auto.py]: why comment?

File: paddlenlp/trainer/training_args.py (hunk 1 above)
Comment (haohongxiang) [reply paddlenlp/trainer/training_args.py]: done

File: llm/llama/run_auto.sh (hunk 2 above)
Comment (haohongxiang) [reply llm/llama/run_auto.sh]: done

File: llm/llama/run_pretrain_auto.py (hunk 3 above)
Comment (haohongxiang) [reply llm/llama/run_pretrain_auto.py]: 这些是冗余代码，已经在前面设置过随机数种子。最新commit中已经删除了

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**